### ISSUE TITLE
Render pinned posts compact/collapsed/consolidated/etc. after interaction

### ISSUE THREAD

**Issue by boscojwho** [mlemgroup/mlem#930]
## Introduction
Briefly: 
- Some communities have many pinned posts (c/mlemapp has 4!). 
- On mobile, users have to scroll past these before interacting with all other posts, even if user has already previously viewed these pinned posts.
- This issue tracks any related discussions relevant to pinned posts UI/UX.
- See branches `/bosco/pinned-posts-ui`, `/bosco/pinned-posts-ui-1` for example implementations.

## Requirements
- [ ] How to make pinned posts more compact, etc.?
- [ ] When to make pinned posts more compact, etc.?
- [ ] Extend post tracker (or the upcoming replacement) to return pinned posts separately.

## Discussion
Reddit introduced a consolidated pinned posts UI on mobile about a year ago to what appears to be a generally negative response from the community (especially from mods):
> https://www.reddit.com/r/ModSupport/comments/ydb66d/consolidated_pinned_posts_on_mobile_is_really/
> https://www.reddit.com/r/redditmobile/comments/15of9h2/ios_2023310_collapsed_pinned_posts_is_ruining/
> https://www.reddit.com/r/redditmobile/comments/15i53bn/ios2023300_pinned_posts_feature_for_subreddits_is/
> https://www.reddit.com/r/modnews/comments/xuvkmy/announcing_consolidated_pinned_posts_on_android/
> https://www.reddit.com/r/ModSupport/comments/xw2jm9/the_new_update_to_pinned_posts_is_already_causing/

Reddit's implementation hides all pinned posts inside a "disclosure group" after two interactions with a community (this may have changed since the original announcement).

Our implementation probably should avoid hiding pinned posts in a disclosure group section, and also allow users to opt-in to auto-collapse/compact etc.

**Comment by boscojwho** [comment]
After reading those Reddit comments, I'm inclined to keep it simple, and do the following:
- Opt-in to auto-collapse as a Settings option.
- (_Possibly_) allow users to choose whether to collapse pinned posts into compact or headline.
- Once opted-in, auto-collapse after viewing pinned post.
- Probably should disable "mark as read on scroll" specifically for pinned posts (i.e. require explicit user interaction before collapsing)?

<img src="https://github.com/mlemgroup/mlem/assets/2549615/55e455bc-e455-46ca-af40-c4d11959f791" width="360">

**Comment by boscojwho** [comment]
Carousel-style pinned posts (`/bosco/pinned-posts-ui-1`)
- I don't recommend this option because it essentially is a variation of the "disclosure group" paradigm, which doesn't seem to be popular (at least on Reddit)
<img src="https://github.com/mlemgroup/mlem/assets/2549615/e89f1e00-1529-4134-bfc2-42faff2a2269" width="320">

**Comment by Sjmarf** [comment]
> After reading those Reddit comments, I'm inclined to keep it simple, and do the following

I agree with this implementation. The carousel takes up less space, but is a little more cumbersome to use IMO, and you also can't use the swipe actions

**Comment by d42ohpaz** [comment]
Can we expand this feature to include the All and Local feeds too for admin pins?

![image](https://github.com/mlemgroup/mlem/assets/2528463/6308572a-8660-44e4-932e-892b02f89cfe)

**Comment by EricBAndrews** [comment]
Closed in #1687
